agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,18 @@
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+import numpy as np
+
+
+class WarmStartReference(WarmStartBase):
+    def generate(self, reference_trajectory: list[TrajectoryPoint]):
+        """
+        Fills the warmstart from the reference
+        Assumes that state `x` is [q, v] and
+        control `u` is initialized to zeros of same shape as v
+        """
+        qs = np.array([point.robot_configuration for point in reference_trajectory])
+        vs = np.array([point.robot_velocity for point in reference_trajectory])
+
+        x_init = np.hstack([qs, vs])
+        u_init = np.zeros_like(vs)
```

Comment:
Here you should use gravity compensation at least no?

Reply:
yeah, depends on what we agree. Gravity compensation would require it to have access to the robot model thus making it more complex. We can start with this since we work with panda that has gravity compensation baked in and create another one later? wdyt

Reply:
Nope panda may have the gravity compensation backed but hits has nothing to do with the warm start.
The solution found by the OCP include the gravity compensation. Hence the warm start should include it as well.
I may not complicate it as much. You can just add a `def setup(pin_model: pin.Model)` method that will be called outside of the mpc class. This way you do not care how this model was created. You just get it and run rnea on the x_init.